<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <meta name="author" content="Ashwin Nanjappa">
  <title>Code Yarns – urlopen got multiple values error</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="../styles.css">
  <!-- RSS feed -->
  <link rel="alternate" type="application/rss+xml" href="https://codeyarns.com/tech/rss.xml" />
  <!-- Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-143311697-1">
  </script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-143311697-1');
  </script>
</head>

<body>

<div class="contentbox">
<div class="header">
    <a href=".." class="header">Code Yarns ‍👨‍💻</a>
  </div>
  <div class="header">
    <a class="header2" href="https://codeyarns.com/tech/">Tech Blog</a> ❖ <a class="header2" href="https://codeyarns.com/personal/">Personal Blog</a>
  </div>
  <div class="header">
      <script async src="https://cse.google.com/cse.js?cx=69d6be64abfa91ed2"> </script>
      <div class="gcse-search"></div>
</div>
</div>

<br />

<div class="contentbox">
<header>
<h1 class="title">urlopen got multiple values error</h1>
<p class="date">📅 2017-Nov-21 ⬩ ✍️ Ashwin Nanjappa ⬩ 🏷️ <a href='index.html#error'>error</a>, <a href='index.html#urllib3'>urllib3</a> ⬩ 📚 <a href="index.html">Archive</a></p>
</header>
<h2 id="problem">Problem</h2>
<p>I tried to run a Python script given by a friend. It ended with this error:</p>
<pre><code>  File &quot;/usr/lib/python2.7/dist-packages/urllib3/request.py&quot;, line 79, in request
    **urlopen_kw)
  File &quot;/usr/lib/python2.7/dist-packages/urllib3/request.py&quot;, line 142, in request_encode_body
    **urlopen_kw)
TypeError: urlopen() got multiple values for keyword argument &#39;body&#39;</code></pre>
<h2 id="solution">Solution</h2>
<p>The script used a Python package, which in turn used <strong>urllib3</strong>. This strange error has nothing to do with my code, but with the urllib3 package. The urllib3 package installed by Ubuntu was pretty old: 1.7.1. Updating to a more recent version of the package will fix the error. Either upgrade using <code>sudo pip install urllib3</code> for system-wide update or update inside your virtualenv.</p>
</div>

<br />

<div class="contentbox">

<div style="text-align: center">
© 2023 Ashwin Nanjappa
•
All writing under <a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA</a> license
•
<a href="https://mastodon.social/@codeyarns">🐘 Mastodon</a>
•
<a href="mailto:codeyarns@gmail.com" style="text-decoration: none; font-size: x-large;">📧 Email</a>
</div>
</div>

</body>
</html>
